### PR TITLE
Fix iOS crash showing paywall for current offering

### LINKED ISSUE #128: Crash when displaying paywall without explicitly providing offering
When showing paywall using this code:

```
val options = PaywallOptions(dismissRequest = { })
Paywall(options)
```
    
the app crashes with:

```
Uncaught Kotlin exception: kotlin.UninitializedPropertyAccessException: lateinit property component has not been initialized
```

When providing offering in the paywall options, it works.

```
var paywallOffering by remember { mutableStateOf<Offering?>(null) }

    LaunchedEffect(Unit) {
        Purchases.configure(getPurchasesApiKey())
        val offerings = Purchases.sharedInstance.awaitOfferings()
        paywallOffering = offerings.current
    }

    if (paywallOffering != null) {
        val options = PaywallOptions(dismissRequest = { onDismiss() }) {
            offering = paywallOffering
            shouldDisplayDismissButton = displayDismissButton
        }
        Paywall(options)
    }
```

1. Environment
    1. Platform: 
    2. SDK version: 1.0.0-beta.1
    3. OS version: 14.5 (23F79)
    4. IDE (e.g. Android Studio, Xcode, Fleet): Android Studio, Xcode
    5. IDE version:
    6. Device and/or emulator/simulator:
        - [x] Device
        - [x] Emulator/simulator
    7. Environment:
        - [x] Closed testing / Sandbox
        - [x] TestFlight
        - [x] Production
2. [Debug logs](https://docs.revenuecat.com/docs/debugging) that reproduce the issue

```
2024-07-11 19:55:13.244710+0200 facememo[24763:8626406] [receipt] INFO: ℹ️ Parsing receipt
2024-07-11 19:55:13.541817+0200 facememo[24763:8626406] [receipt] INFO: ℹ️ Receipt parsed successfully
Uncaught Kotlin exception: kotlin.UninitializedPropertyAccessException: lateinit property component has not been initialized
    at 0   facememo                            0x1014cb60f        kfun:kotlin.Throwable#<init>(kotlin.String?){} + 119 
    at 1   facememo                            0x1014c4b1b        kfun:kotlin.Exception#<init>(kotlin.String?){} + 115 
    at 2   facememo                            0x1014c4d3b        kfun:kotlin.RuntimeException#<init>(kotlin.String?){} + 115 
    at 3   facememo                            0x1014c5c83        kfun:kotlin.UninitializedPropertyAccessException#<init>(kotlin.String?){} + 115 
    at 4   facememo                            0x1015022c3        kfun:kotlin.native.internal#ThrowUninitializedPropertyAccessException(kotlin.String){}kotlin.Nothing + 367 
    at 5   facememo                            0x10227991b        kfun:androidx.compose.ui.interop.EmbeddedInteropComponent.<get-component>#internal + 199 
    at 6   facememo                            0x10227d04f        kfun:androidx.compose.ui.interop.UIKitViewController$lambda$26$lambda$25#internal + 383 
    at 7   facememo                            0x10227fd63        kfun:androidx.compose.ui.interop.$UIKitViewController$lambda$26$lambda$25$FUNCTION_REFERENCE$30.invoke#internal + 79 
    at 8   facememo                            0x10227fe8b        kfun:androidx.compose.ui.interop.$UIKitViewController$lambda$26$lambda$25$FUNCTION_REFERENCE$30.$<bridge-UNN>invoke(){}#internal + 71 
    at 9   facememo                            0x101606ae7        kfun:kotlin.Function0#invoke(){}1:0-trampoline + 99 
    at 10  facememo                            0x1022f5173        kfun:androidx.compose.ui.window.MetalRedrawer.draw$lambda$4#internal + 2819 
    at 11  facememo                            0x1022f685f        kfun:androidx.compose.ui.window.MetalRedrawer.$draw$lambda$4$FUNCTION_REFERENCE$4.invoke#internal + 103 
    at 12  facememo                            0x1022f6a4b        kfun:androidx.compose.ui.window.MetalRedrawer.$draw$lambda$4$FUNCTION_REFERENCE$4.$<bridge-UNN>invoke(){}#internal + 71 
    at 13  facememo                            0x101606ae7        kfun:kotlin.Function0#invoke(){}1:0-trampoline + 99 
    at 14  facememo                            0x1022f2473        kfun:androidx.compose.ui.window.MetalRedrawer.draw#internal + 7271 
    at 15  facememo                            0x1022f4193        kfun:androidx.compose.ui.window.MetalRedrawer.<init>$lambda$0#internal + 435 
    at 16  facememo                            0x1022f62c7        kfun:androidx.compose.ui.window.MetalRedrawer.$<init>$lambda$0$FUNCTION_REFERENCE$0.invoke#internal + 71 
    at 17  facememo                            0x1022f6397        kfun:androidx.compose.ui.window.MetalRedrawer.$<init>$lambda$0$FUNCTION_REFERENCE$0.$<bridge-UNN>invoke(){}#internal + 71 
    at 18  facememo                            0x101606ae7        kfun:kotlin.Function0#invoke(){}1:0-trampoline + 99 
    at 19  facememo                            0x1022f70df        kfun:androidx.compose.ui.window.DisplayLinkProxy.handleDisplayLinkTick#internal + 151 
    at 20  facememo                            0x1022f7193        kfun:androidx.compose.ui.window.DisplayLinkProxy.$imp:handleDisplayLinkTick#internal + 143 
    at 21  QuartzCore                          0x1998b7ecb        <redacted> + 47 
    at 22  QuartzCore                          0x1998b622f        <redacted> + 875 
    at 23  QuartzCore                          0x1998b5dcf        <redacted> + 351 
    at 24  UIKitCore                           0x19a4ad76b        <redacted> + 83 
    at 25  UIKitCore                           0x19a4ad3af        <redacted> + 171 
    at 26  UIKitCore                           0x19a4ae253        <redacted> + 91 
    at 27  CoreFoundation                      0x1981cb833        <redacted> + 27 
    at 28  CoreFoundation                      0x1981cb7c7        <redacted> + 175 
    at 29  CoreFoundation                      0x1981c9297        <redacted> + 243 
    at 30  CoreFoundation                      0x1981c8483        <redacted> + 827 
    at 31  CoreFoundation                      0x1981c7cd7        CFRunLoopRunSpecific + 607 
    at 32  GraphicsServices                    0x1dd0781a7        GSEventRunModal + 163 
    at 33  UIKitCore                           0x19a80090b        <redacted> + 887 
    at 34  UIKitCore                           0x19a8b49cf        UIApplicationMain + 339 
    at 35  SwiftUI                             0x19c3b8147        <redacted> + 414603 
    at 36  SwiftUI                             0x19c364713        <redacted> + 72023 
    at 37  SwiftUI                             0x19c3704cf        <redacted> + 120595 
    at 38  facememo                            0x100fccac3        $s8facememo0A3AppV5$mainyyFZ + 39 
    at 39  facememo                            0x100fccb6f        main + 11 
    at 40  dyld                                0x1bb879e4b        <redacted> + 2239
```
    
3. Expected behavior
Paywall should fetch offering implicitly, like it works on android. If that's not possible for whatever reason, we should make the offering argument in the paywall options object to be mandatory.
Documentation also shows code that doesn't fetch offerings: [docs](https://www.revenuecat.com/docs/tools/paywalls/displaying-paywalls#how-to-display-a-fullscreen-paywall-in-your-app-4)

### PR DESCRIPTION
**Pull request by JayShortway**
As the title says. The commit messages are pretty descriptive. The actual fix is tiny and in the last commit.

Closes #128

### CODE REVIEW

File: iosApp/Podfile.lock [hunk 1]
```diff
@@ -1,16 +1,16 @@
 PODS:
```
Comment (aboedo) [inline iosApp/Podfile.lock:1]: we may not even wanna commit this file, it'll fall out of date super often

**Review by aboedo — APPROVED** [review]

File: iosApp/Podfile [hunk 2]
```diff
@@ -6,6 +6,6 @@ target 'iosApp' do
   # use_frameworks!
 
   # Pods for iosApp
-  pod 'PurchasesHybridCommon'
-  pod 'PurchasesHybridCommonUI'
+  pod 'PurchasesHybridCommon', '11.1.1'
+  pod 'PurchasesHybridCommonUI', '11.1.1'
```
Comment (tonidero) [inline iosApp/Podfile:10]: Do we need to add these to the update PHC fastlane command: https://github.com/RevenueCat/purchases-kmp/blob/d85257604bc8ae57b33f0df559d450c9491ebb04/fastlane/Fastfile#L79

**Review by tonidero — APPROVED** [review]

File: iosApp/Podfile.lock (hunk 1 above)
Comment (JayShortway) [reply iosApp/Podfile.lock:1]: I see indeed that other hybrids don't have it version controlled either (although PHC does). In general I think it's nice to be a bit more explicit about transitive dependencies, so I'd propose to remove the file once it becomes a problem.

File: iosApp/Podfile (hunk 2 above)
Comment (JayShortway) [reply iosApp/Podfile:10]: Yesss we should, thanks! I thought we'd need to make a change to the PHC repo, but it's right here of course. Done: https://github.com/RevenueCat/purchases-kmp/pull/129/commits/ff83102fa017026ed60ac636c763e049be28e928

File: fastlane/Fastfile [hunk 3]
```diff
@@ -80,6 +80,8 @@ lane :update_hybrid_common do |options|
     "gradle/libs.versions.toml" => ["revenuecat-common = \"{x}\""],
     "core/core.podspec" => ["spec.dependency 'PurchasesHybridCommon', '{x}'"],
     "revenuecatui/revenuecatui.podspec" => ["spec.dependency 'PurchasesHybridCommonUI', '{x}'"],
+    "iosApp/Podfile" => ["  pod 'PurchasesHybridCommon', '{x}'"],
+    "iosApp/Podfile" => ["  pod 'PurchasesHybridCommonUI', '{x}'"],
```
Comment (tonidero) [inline fastlane/Fastfile:84]: I think you can add them both within the same array
```suggestion
    "iosApp/Podfile" => ["  pod 'PurchasesHybridCommon', '{x}'", "  pod 'PurchasesHybridCommonUI', '{x}'"],
```

Comment (JayShortway) [reply fastlane/Fastfile]: Even better, thanks! https://github.com/RevenueCat/purchases-kmp/pull/129/commits/e8e18dde5f398d91b738e6550eb114984aff9ef1

**Pull request merged**